src/planner/ostree.rs

```diff
@@ -0,0 +1,406 @@
+                [Service]\n\
+                Type=oneshot\n\
+                ExecStartPre=chattr -i /\n\
+                ExecStart=mkdir -p /nix\n\
+                ExecStopPost=chattr +i /\n\
+            "
+        .to_string();
+        let nix_directory_unit = CreateFile::plan(
+            "/etc/systemd/system/nix-directory.service",
+            None,
+            None,
+            0o0644,
+            nix_directory_buf,
+            false,
+        )
+        .await
+        .map_err(PlannerError::Action)?;
+        plan.push(nix_directory_unit.boxed());
+
+        let create_bind_mount_buf = format!(
+            "\
+                [Unit]\n\
+                Description=Mount `{persistence}` on `/nix`\n\
+                PropagatesStopTo=nix-daemon.service\n\
+                PropagatesStopTo=nix-directory.service\n\
+                After=nix-directory.service\n\
+                Requires=nix-directory.service\n\
+                ConditionPathIsDirectory=/nix\n\
+                DefaultDependencies=no\n\
+                \n\
+                [Mount]\n\
+                What={persistence}\n\
+                Where=/nix\n\
+                Type=none\n\
+                DirectoryMode=0755\n\
+                Options=bind\n\
+                \n\
+                [Install]\n\
+                RequiredBy=nix-daemon.service\n\
+                RequiredBy=nix-daemon.socket\n
+            ",
+            persistence = self.persistence.display(),
+        );
+        let create_bind_mount_unit = CreateFile::plan(
+            "/etc/systemd/system/nix.mount",
+            None,
+            None,
+            0o0644,
+            create_bind_mount_buf,
+            false,
+        )
+        .await
+        .map_err(PlannerError::Action)?;
+        plan.push(create_bind_mount_unit.boxed());
+
+        let ensure_symlinked_units_resolve_buf = "\
+        [Unit]\n\
+        Description=Ensure Nix related units which are symlinked resolve\n\
+        After=nix.mount\n\
+        Requires=nix.mount\n\
+        DefaultDependencies=no\n\
+        \n\
+        [Service]\n\
+        Type=oneshot\n\
+        RemainAfterExit=yes\n\
+        ExecStart=/usr/bin/systemctl daemon-reload\n\
+        ExecStart=/usr/bin/systemctl restart --no-block nix-daemon.socket\n\
+        \n\
+        [Install]\n\
+        WantedBy=sysinit.target\n\
+    "
+        .to_string();
+        let ensure_symlinked_units_resolve_unit = CreateFile::plan(
+            "/etc/systemd/system/ensure-symlinked-units-resolve.service",
+            None,
+            None,
+            0o0644,
+            ensure_symlinked_units_resolve_buf,
+            false,
+        )
+        .await
+        .map_err(PlannerError::Action)?;
+        plan.push(ensure_symlinked_units_resolve_unit.boxed());
+
+        // We need to remove this path since it's part of the read-only install.
+        let mut shell_profile_locations = ShellProfileLocations::default();
```

Comment:
I wonder if we should make `ShellProfileLocations::default()` check if the items if candidate paths are read-only and exclude them...

Reply:
I played a bit with this, but couldn't find a good way to do this.
One alternative would be to stop erroring when this files cannot be created and logging warnings instead.

Reply:
Let's punt it for now. :)

Reply:
Not sure what that means 😅